unit_scaling/functional.py

```diff
@@ -252,6 +276,47 @@ def embedding(
     )
 
 
+if hasattr(F, "scaled_dot_product_attention"):
+
+    @docstring_from(
+        F.scaled_dot_product_attention,
+        short_description=(
+            "A **unit-scaled** dot-product attention function. Note that this will use"
+            "whatever underlying PyTorch scaled_dot_product_attention implementation"
+            "is available, so if flash attention is enabled it will be used here too."
+            "\n\n"
+            "Computes scaled dot product attention on query, key and value tensors,"
+            "using an optional attention mask if passed, and applying dropout if a"
+            "probability greater than 0.0 is specified."
+        ),
+    )
+    def scaled_dot_product_attention(
+        query: Tensor,
+        key: Tensor,
+        value: Tensor,
+        attn_mask: Optional[Tensor] = None,
+        dropout_p: float = 0.0,
+        is_causal: bool = False,
```

Comment:
Should the docs include a warning that unit-scaling is only approximate when using `attn_mask`, `dropout_p` and `is_causal`?

Reply:
I'm tempted to say that given there's a pretty wide range of cases where values only come out approximately unit-scaled, a warning here would mean lots of warnings - for instance, probably the largest thing that pushes us from true US is the correlation introduced by this op. But I'd rather not go into those details at this stage - at some point would be nice to address this more generally in the user guide?

Reply:
Yes, agree that this approach sounds fair.